cmd/zetaclientd/start.go

```diff
@@ -210,6 +213,12 @@ func start(_ *cobra.Command, _ []string) error {
 	// Set P2P ID for telemetry
 	telemetryServer.SetP2PID(server.GetLocalPeerID())
 
+	// Create a notification context for background threads.These threads are responsible for sending shutdown signals to the main thread.
+	notifyCtx, cancel := context.WithCancelCause(ctx)
+
+	// Start background threads
+	defer startBackgroundThreads(notifyCtx, cancel, zetacoreClient, masterLogger)
```

Comment:
Why using defer here?

Reply:
The defer is to stop the background threads in case of a shutdown via os signals. I don't think its mandatory to have this , but can be good to have